openshift-install-powervs

```diff
@@ -1287,29 +1372,23 @@ function main {
     "-var")
       shift
       var="$1"
-      #TODO: Need validation on variable key=value, currently it is only checking if equals sign is present or not
-      [[ "$var" != *"="* ]] && error "The given -var option must be a variable name and value separated by an equals sign, eg: -var \"key=value\""
+      [[ $(echo "$var" | grep -o "[[:blank:]]" | wc -l) -gt 0 ]] && error "The -var '$var' option cannot have whitespaces in it. Please use -var-file instead."
+      [[ "$var" != *"="* ]] && error "The given -var '$var' option must be a key-value pair separated by an equals sign, eg: -var \"key=value\""
       vars+=" -var $var"
-      SERVICE_INSTANCE_ID=$(echo "$var" | grep "service_instance_id" | cut -d '=' -f 2)
-      debug_switch
-      VAR_IBMCLOUD_API_KEY=$(echo "$var" | grep "ibmcloud_api_key" | cut -d '=' -f 2)
-      debug_switch
-      VAR_PULL_SECRET_FILE=$(echo "$var" | grep "pull_secret_file" | cut -d '=' -f 2)
-      VAR_PRIVATE_KEY_FILE=$(echo "$var" | grep "private_key_file" | cut -d '=' -f 2)
-      VAR_PUBLIC_KEY_FILE=$(echo "$var" | grep "public_key_file" | cut -d '=' -f 2)
+      process_var
       ;;
     "-var-file")
       shift
       varfile="$1"
       [[ ! -s "$varfile" ]] && error "File $varfile does not exist"
       vars+=" -var-file ../$varfile"
-      SERVICE_INSTANCE_ID=$(read_varfile "service_instance_id")
-      debug_switch
-      VAR_IBMCLOUD_API_KEY=$(read_varfile "ibmcloud_api_key")
-      debug_switch
-      VAR_PULL_SECRET_FILE=$(read_varfile "pull_secret_file")
-      VAR_PRIVATE_KEY_FILE=$(read_varfile "private_key_file")
-      VAR_PUBLIC_KEY_FILE=$(read_varfile "public_key_file")
+      process_varfile "service_instance_id"
```

Comment:
Can you please make an array of all the variable names and use a for loop in `process_varfile`?

Reply:
Also... I see you have case in `process_varfile` then multiple calls are not needed here again.

Reply:
Resolved